reducer/src/main/java/com/graphicsfuzz/reducer/reductionopportunities/LiteralToUniformReductionOpportunity.java

```diff
@@ -25,18 +30,40 @@
 
   private final Expr literalExpr;
   private final PipelineInfo pipelineInfo;
+  private final TranslationUnit translationUnit;
+  private final int index;
 
-  LiteralToUniformReductionOpportunity(Expr literalExpr,
+  LiteralToUniformReductionOpportunity(TranslationUnit translationUnit,
+                                       Expr literalExpr,
+                                       int index,
                                        PipelineInfo pipelineInfo,
                                        VisitationDepth depth) {
     super(depth);
+
+    this.index = index;
     this.literalExpr = literalExpr;
     this.pipelineInfo = pipelineInfo;
+    this.translationUnit = translationUnit;
+  }
+
+  public Expr getLiteral() {
+    return this.literalExpr;
+  }
+
+  public int getIndex() {
+    return this.index;
   }
 
   @Override
   void applyReductionImpl() {
-    // TODO: ...
+
+    IParentMap parentMap = IParentMap.createParentMap(translationUnit);
+
+    ArrayIndexExpr aie = new ArrayIndexExpr(new VariableIdentifierExpr("values"),
```

Comment:
The name "values" might already be defined.  Please add a new constant to Constants, with a suitable name (using the "_GLF_" prefix).

Reply:
Done.